### DISCUSSION TITLE
Add AOP capabilities

### DISCUSSION THREAD

**Discussion by joshfree** [dotnet/runtime#72038] [Ideas]
_From @pleandre on May 3, 2018 14:41_

Hello all,

Currently their are not many ways to do AOP (Aspect Oriented Programming) in .net core. The only way I see is to use libraries like Postsharp / Fody which adds some compilation steps ([MSIL rewritting](http://doc.postsharp.net/how-it-works)) and complex plumbing.

I think it would be great to have some functionalities embedded in .net core for aspect weaving.

It would make it more simple to create some custom attributes and logic (and be able to use it like annotations in java / spring aop / aspectJ) 
e.g. be able to develop a [Schedule], a [Caching], a [RegisterToEventBus] attribute easily with custom logic and injecting code at compilation time on methods, class, etc. tagged (adding code before a method runs, after, on return value, on a getter, setter, exception, etc.. at compilation time)

Is it something you think should be handled by dotnet core ?

Thank you,
Pierre

_Copied from original issue: dotnet/core#1495_

**Comment by daveaglick** [comment]
FWIW, build-time code generation (which I would consider AOP a subset of use cases for) is discussed at length in https://github.com/dotnet/roslyn/issues/19505

**Comment by pmgriffin** [comment]
Close one issue referencing another and then close the referenced issue as a dupe ?  :)

**Comment by athinboy** [comment]
AOP is so import , especially for distribute system log、performance  trace、hotswap、Auto DB transaction Manage,  etc.
I hope .net can implement all  features that has been implemented in java , or  more and more project will turn to java .
Thanks !

**Comment by Kebechet** [comment]
Having AOP would simplify our source in so many ways. Is there any update ?
